server/sonar-server/src/test/java/org/sonar/server/project/ws/GhostsActionTest.java

```diff
@@ -35,137 +37,203 @@
 import org.sonar.db.component.SnapshotTesting;
 import org.sonar.db.organization.OrganizationDto;
 import org.sonar.server.exceptions.ForbiddenException;
+import org.sonar.server.organization.DefaultOrganizationProvider;
+import org.sonar.server.organization.TestDefaultOrganizationProvider;
 import org.sonar.server.tester.UserSessionRule;
-import org.sonar.server.ws.WsTester;
-import org.sonar.test.JsonAssert;
+import org.sonar.server.ws.TestResponse;
+import org.sonar.server.ws.WsActionTester;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.db.component.SnapshotDto.STATUS_PROCESSED;
+import static org.sonar.db.component.SnapshotDto.STATUS_UNPROCESSED;
+import static org.sonar.test.JsonAssert.assertJson;
 
 public class GhostsActionTest {
 
   @Rule
   public DbTester db = DbTester.create(System2.INSTANCE);
-
   @Rule
   public UserSessionRule userSessionRule = UserSessionRule.standalone();
 
-  DbClient dbClient = db.getDbClient();
-  WsTester ws = new WsTester(new ProjectsWs(new GhostsAction(dbClient, userSessionRule)));
+  private DefaultOrganizationProvider defaultOrganizationProvider = TestDefaultOrganizationProvider.from(db);
+  private DbClient dbClient = db.getDbClient();
+  private WsActionTester underTest = new WsActionTester(new GhostsAction(dbClient, userSessionRule, defaultOrganizationProvider));
 
   @Test
-  public void ghost_projects_without_analyzed_projects() throws Exception {
-    userSessionRule.setGlobalPermissions(UserRole.ADMIN);
-    insertNewGhostProject("1");
-    insertNewGhostProject("2");
-    insertNewActiveProject("3");
+  public void verify_definition() {
+    WebService.Action action = underTest.getDef();
+    assertThat(action.description()).isEqualTo("List ghost projects.<br /> Requires 'Administer System' permission.");
+    assertThat(action.since()).isEqualTo("5.2");
+    assertThat(action.isInternal()).isFalse();
+
+    assertThat(action.params()).hasSize(5);
+
+    Param organization = action.param("organization");
+    assertThat(organization.description()).isEqualTo("the organization key");
+    assertThat(organization.since()).isEqualTo("6.3");
+    assertThat(organization.isRequired()).isFalse();
+    assertThat(organization.isInternal()).isTrue();
+  }
 
-    WsTester.Result result = ws.newGetRequest("api/projects", "ghosts").execute();
+  @Test
+  public void ghost_projects_without_analyzed_projects() throws Exception {
```

Comment:
I suggest also to add a test to verify that projects from other organizations are not accessed by unauthorized users.